test/realm_test.dart

```diff
@@ -1898,6 +1899,29 @@ Future<void> main([List<String>? args]) async {
     expect(query.length, 1);
     expect(query[0].name, productName);
   });
+
+  test('Realm logger change level to error', () {
+    final oldLogger = Realm.logger;
+    try {
+      int count = 0;
```

Comment:
Do you want to actually test something regarding count at the end? As is it could be the callback was never invoked.